### PR TITLE
vmselect:fix init sn take too much time

### PR DESCRIPTION
**Pull request by Haleygo**
fix https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4364

### CODE REVIEW

**Comment by hagen1778** [comment]
@f41gh7 @Amper would you mind taking a look?

File: app/vmselect/netstorage/netstorage.go [hunk 1]
```diff
@@ -2698,44 +2698,53 @@ func initStorageNodes(addrs []string) *storageNodesBucket {
...
-			tenantsRequests:             ms.NewCounter(fmt.Sprintf(`vm_requests_total{action="tenants", type="rpcClient", name="vmselect", addr=%q}`, addr)),
-			tenantsErrors:               ms.NewCounter(fmt.Sprintf(`vm_request_errors_total{action="tenants", type="rpcClient", name="vmselect", addr=%q}`, addr)),
-
-			metricBlocksRead: ms.NewCounter(fmt.Sprintf(`vm_metric_blocks_read_total{name="vmselect", addr=%q}`, addr)),
-			metricRowsRead:   ms.NewCounter(fmt.Sprintf(`vm_metric_rows_read_total{name="vmselect", addr=%q}`, addr)),
-		}
-		sns = append(sns, sn)
+		wg.Add(1)
+		go func(addr string) {
+			if _, _, err := net.SplitHostPort(addr); err != nil {
```
Comment (f41gh7) [inline app/vmselect/netstorage/netstorage.go:2707]: I'd recommend put `defer wg.Done()` at first line of goroutine function. It prevents possible bugs with fast path and improves readability.

**Review by Amper — APPROVED** [review]

File: app/vmselect/netstorage/netstorage.go [hunk 2]
```diff
@@ -2698,44 +2698,53 @@ func initStorageNodes(addrs []string) *storageNodesBucket {
 		logger.Panicf("BUG: addrs must be non-empty")
 	}
 	sns := make([]*storageNode, 0, len(addrs))
+	var snsLock sync.Mutex
```
[the remark this answers is not in the document]
Comment (Haleygo) [reply app/vmselect/netstorage/netstorage.go:2700]: Sure, fixed.

File: app/vmselect/netstorage/netstorage.go (hunk 1 above)
Comment (Haleygo) [reply app/vmselect/netstorage/netstorage.go:2708]: OK, I saw there is no obvious way to break this routine, so didn't use defer.
Fixed now.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**